src/Compilers/CSharp/Portable/Binder/PatternExplainer.cs

```diff
@@ -114,50 +190,79 @@ int distance(BoundDecisionDagNode x)
             unnamedEnumValue = false;
 
             // Compute the path to the node, excluding the node itself.
-            var pathToNode = ShortestPathToNode(nodes, targetNode, nullPaths, out requiresFalseWhenClause);
+            var shortestPathToNode = ShortestPathToNode(nodes, targetNode, nullPaths, out requiresFalseWhenClause);
+            gatherConstraintsAndEvaluations(targetNode, shortestPathToNode, out var constraints, out var evaluations);
+
+            try
+            {
+                return SamplePatternForTemp(rootIdentifier, constraints, evaluations, requireExactType: false, ref unnamedEnumValue);
+            }
+            catch (NoRemainingValuesException)
+            {
+            }
 
-            var constraints = new Dictionary<BoundDagTemp, ArrayBuilder<(BoundDagTest, bool)>>();
-            var evaluations = new Dictionary<BoundDagTemp, ArrayBuilder<BoundDagEvaluation>>();
-            for (int i = 0, n = pathToNode.Length; i < n; i++)
+            // In rare cases, the shortest path isn't the one that yields a sample
+            foreach (var (altPathToNode, altRequiresFalseWhenClause) in GetPathsToNode(nodes, targetNode, nullPaths))
             {
-                BoundDecisionDagNode node = pathToNode[i];
-                switch (node)
+                requiresFalseWhenClause = altRequiresFalseWhenClause;
+                gatherConstraintsAndEvaluations(targetNode, altPathToNode, out constraints, out evaluations);
+
+                try
                 {
-                    case BoundTestDecisionDagNode t:
-                        {
-                            BoundDecisionDagNode nextNode = (i < n - 1) ? pathToNode[i + 1] : targetNode;
-                            bool sense = t.WhenTrue == nextNode || (t.WhenFalse != nextNode && t.WhenTrue is BoundWhenDecisionDagNode);
-                            BoundDagTest test = t.Test;
-                            BoundDagTemp temp = test.Input;
-                            if (test is BoundDagTypeTest && sense == false)
-                            {
-                                // A failed type test is not very useful in constructing a counterexample,
-                                // at least not without discriminated unions, so we just drop them.
-                            }
-                            else
+                    return SamplePatternForTemp(rootIdentifier, constraints, evaluations, requireExactType: false, ref unnamedEnumValue);
```

Comment:
>unnamedEnumValue [](http://example.com/codeflow?start=119&length=16)

Consider setting this to false before calling ```SamplePatternForTemp``` to match what we do for the first attempt